test/session_tests/DocumentStoreTest.ts

```diff
@@ -182,6 +182,16 @@ describe('Document store test', () => {
       expect(product['@metadata']['Raven-Node-Type']).to.not.exist;  
       expect(product['@metadata']['@collection']).to.equals('@empty');      
     });
+
+    it('shouldn\'t set id on plain object without id attribute defined', async () => {
+      let product: Product = <Product>{name: "New Product"};    
+
+      session = store.openSession({requestExecutor});      
+      await session.store<Product>(product);
+      await session.saveChanges();
+
+      expect(product.id).to.be.empty;      
```

Comment:
Yes that's the other side of it. JS properties are more flexible on this matter than C#, maybe it's worth it to treat them in a different way. All right, let's add the property if it's not there @a-serdukov-mobilunity